### DISCUSSION TITLE
Adding New Characeters To Text Recog models

### DISCUSSION THREAD

**Discussion by TheSeriousProgrammer** [open-mmlab/mmocr#1499] [Q&A] [answered]
I am currenly using CRNN's text recog model, but it doesnt have a few special charecters which I need such as ("?","/") and a few others
How can I calibrate CRNN and modify my inference code to support these new charecters?

Thank  you so much for the impressive framework though

**Comment by gaotongxiao** [comment, marked as answer]
Similar to https://github.com/open-mmlab/mmocr/discussions/1488, it depends on the version that you're using.

## MMOCR 0.x

You can control the supported character set by the parameter `dict_type` here:

https://github.com/open-mmlab/mmocr/blob/3188e534042096e27dbd751deb83df8dedc34314/configs/_base_/recog_models/crnn.py#L1-L2

We have a list of pre-defined DICTs, where `DICT90` shall satisfy your requirement:

https://github.com/open-mmlab/mmocr/blob/37833ada4bbd951c55702cbb52b192358c705dff/mmocr/models/textrecog/convertors/base.py#L24-L26

## MMOCR 1.0

It can be controlled via the `dict_file` parameter:

https://github.com/open-mmlab/mmocr/blob/f30c16ce96bd2393570c04eeb9cf48a7916315cc/configs/textrecog/crnn/_base_crnn_mini-vgg.py#L1-L4

MMOCR 1.0 has provided some commonly used dictionary files, including [english_digits_symbols](https://github.com/open-mmlab/mmocr/blob/1.x/dicts/english_digits_symbols.txt) that works for your case.

**Reply by TheSeriousProgrammer** [reply]
Thank you that's what I was looking for , will check it out and comeback

**Reply by TheSeriousProgrammer** [reply]
It worked perfectly , thanks for the response
